File: internal/models/team.go
package models

import "time"

type Team struct {
	ID        string    `json:"id"`
	OwnerID   string    `json:"owner_id"`
	Name      string    `json:"name"`
	UpdatedAt time.Time `json:"updated_at"`
	CreatedAt time.Time `json:"created_at"`
}

type TeamUser struct {
	ID        string    `json:"id"`
	TeamID    string    `json:"team_id"`
	UserID    string    `json:"user_id"`
	UpdatedAt time.Time `json:"updated_at"`
	CreatedAt time.Time `json:"created_at"`
}

type TeamInvite struct {
	ID        string    `json:"id"`
	TeamID    string    `json:"team_id"`
	InvitedBy string    `json:"invited_by"`
	InvitedTo string    `json:"invited_to"`
	CreatedAt time.Time `json:"created_at"`
}

File: website/src/lib/schemas/team.ts
// Team type
export interface Team {
	id: string;
	owner_id: string;
	name: string;
	updated_at: string;
	created_at: string;
}

// Team User type
export interface TeamUser {
	id: string;
	team_id: string;
	user_id: string;
	updated_at: string;
	created_at: string;
}

// Team Invite type
export interface TeamInvite {
	id: string;
	team_id: string;
	invited_by: string;
	invited_to: string;
	updated_at: string;
	created_at: string;
}
